### DISCUSSION TITLE
question about wait()

### DISCUSSION THREAD

**Discussion by AsafDin** [ElSnoMan/pyleniumio#169] [Q&A] [answered]
py.get('#userCode').type('user')
        py.get('#password').type('password')
        #hpy.get('localName=button').click()
        exceptions = [NoSuchElementException, WebDriverException]
        py.get('button.red-coloring-btn.login-btn').click()
       py.wait(ignored_exceptions=exceptions).until(ec.title_is('דף  הבית'))

Hi I'm using the above code to login to a site.
after the click is executed a new page is loaded with the title as the expected but the test fails with timeout on the wait.
the following is from the loaded page:
<head>
  <meta charset="utf-8">
  <title>דף  הבית</title>
  <base href="/ng-portals/rb/he/">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="icon" type="image/x-icon" href="favicon.ico">

I want the test to wait for the page to fully load before it continues with other steps, I tried to use the title , other elements in the page but it always the same result. what am I missing ?

**Comment by ElSnoMan** [comment, marked as answer]
Provide the error that is being raised. The more information you provide, the easier it is to debug and provide guidance.

From looking at the provided code, line 3 isn't going to work since `localName=button` is not valid CSS. It looks like it's commented out, but it's hard to tell since it's not formatted.

Pylenium waits for a lot of things automatically, so you probably don't need to wait for the title of the page and just interact with the element you care about. That's my first bit of advice.

But, if you do care about the title, then use a `.should()` expectation which does a wait as well

```python
py.should().contain_title('בית')
```
